pymrio/tools/iobalance.py

```diff
@@ -0,0 +1,189 @@
+        num_rows_Z, num_cols_Z = Inter_industry.shape
+        if num_rows_Z == num_cols_Z:
+            pass
+        else:
+            raise NotSquareDataFrameError("Number of rows does not equal columns")
+    else:
+        pass
+    
+    #Check shape of other matrices
+    
+    #Value Added
+    if isinstance(value_added, pd.DataFrame):
+        num_rows_VA, num_cols_VA = value_added.shape
+        if num_cols_VA == num_cols_Z:
+            pass
+        else:
+            raise MatrixNotCompatibleError("Value Added shape is not compatible with InterIndustry Matrix")
+    else:
+        pass    
+    
+    #Check Final Demand
+    if isinstance(final_demand, pd.DataFrame):
+        num_rows_Y, num_cols_Y = final_demand.shape
+        if num_rows_Y == num_rows_Z:
+            pass
+        else:
+            raise MatrixNotCompatibleError("Final Demand shape is not compatible with InterIndustry Matrix")
+    else:
+        pass    
+        
+    
+    #Check Total Output
+    if isinstance(total_output, pd.DataFrame):
+        num_rows_X, num_cols_X = total_output.shape
+        if num_rows_X == num_rows_Z:
+            pass
+        else:
+            raise MatrixNotCompatibleError("Total Output shape is not compatible with InterIndustry Matrix")
+    else:
+        pass
+
+    print("Matrices Shapes are compatible")
+    
+    
+    #check if total Output is the sum of Industry and final demand outputs
+    for i in range(len(Inter_industry)):
+        
+        
+        Inter_industry_sum = Inter_industry.iloc[i].sum()
+        Final_demand_sum = final_demand.iloc[i].sum() 
+        total= Inter_industry_sum+Final_demand_sum
+        
+        
+        if round(total) == round(float(total_output.iloc[i])):
+            pass
+        else:
+            raise TotalOutputError('Total output does not equal sum of intermediate and final demand consumption')
+    
+    #check if system totals are balanced by summing total output and value added + InterIndustry
+    
+    X_sum = total_output.sum(axis=0) #Total Output sum
+    Z_sum_vertical = Inter_industry.sum(axis=0).sum() #Input Matrix sum (vertical sum of InterIndustry)
+    VA_sum = value_added.sum(axis=0).sum() #Value Added sum
+    
+    Input = Z_sum_vertical + VA_sum
+    Output = X_sum 
+    
+    #round to the nearest thousands decimal
+    if (Input/Output < 1.01).all() and (Input/Output > 0.99).all():
+        pass
+    else:
+        raise SystemNotBalancedError('The IO System is not Balanced: Sum of Inputs does not Equal sum of outputs')
+    
+    print('IO System is Balanced: Total Inputs equal Total Outputs')
+    
+    
+    #check if each sector in each region is balanced
+    for i in range(len(Inter_industry)):    
+        
+        #first check if the value added is multi-dimensional         
+        if value_added.ndim> 1:
+            #if yes then sum vertically and return a single dimensional array
+            value_added =value_added.sum(axis=0)
```

Comment:
please run the function through a linter, we use black in pymrio.
More info in the "contribution.rst" in the root